### PR TITLE
added chatgpt (chat-completions) API

### PR DESCRIPTION
**Pull request by bashar94**
Added Official ChatGPT API (chat-completions) [https://platform.openai.com/docs/guides/chat](https://platform.openai.com/docs/guides/chat)

### CODE REVIEW

**Comment by bashar94** [comment]
@orhanerday please check

File: README.md [hunk 1]
```diff
@@ -419,6 +419,26 @@ $classification = $open_ai->classification([
...
+
+Given a chat conversation, the model will return a chat completion response.
+
+ ```php
+$complete = $open_ai->chat([
+    'model' => 'gpt-3.5-turbo',
+    'messages' => [
+        [
+            "role" => "user", 
+            "content": "Hello!"
```
Comment (oliverschloebe) [inline README.md:432]: Should be `"content" => "Hello!"`

Comment (bashar94) [reply README.md]: fixed it

**Comment by df2012** [comment]
I tried the solution and I've been getting this error when trying to use it "Call to undefined method Orhanerday\OpenAi\OpenAi::baseUrl()".

"
    public function chat($opts, $stream = null)
    {
        if ($stream != null && array_key_exists('stream', $opts)) {
            if (! $opts['stream']) {
                throw new Exception(
                    'Please provide a stream function. Check https://github.com/orhanerday/open-ai#stream-example for an example.'
                );
            }
            $this->stream_method = $stream;
        }
        $opts['model'] = $opts['model'] ?? $this->model;
        $url = Url::chatUrl();
        $this->baseUrl($url);
        return $this->sendRequest($url, 'POST', $opts);
    }
"

Can you help me?

**Comment by bashar94** [comment]
> I tried the solution and I've been getting this error when trying to use it "Call to undefined method Orhanerday\OpenAi\OpenAi::baseUrl()".
> 
> " public function chat($opts, $stream = null) { if ($stream != null && array_key_exists('stream', $opts)) { if (! $opts['stream']) { throw new Exception( 'Please provide a stream function. Check https://github.com/orhanerday/open-ai#stream-example for an example.' ); } $this->stream_method = $stream; } $opts['model'] = $opts['model'] ?? $this->model; $url = Url::chatUrl(); $this->baseUrl($url); return $this->sendRequest($url, 'POST', $opts); } "
> 
> Can you help me?

@df2012 would you please pull the code, I pushed another small fix but seems like it is not related to your issue. please check if you have `baseUrl()` function at the end of `src/OpenAi.php` file

**Comment by df2012** [comment]
> > I tried the solution and I've been getting this error when trying to use it "Call to undefined method Orhanerday\OpenAi\OpenAi::baseUrl()".
> > " public function chat($opts, $stream = null) { if ($stream != null && array_key_exists('stream', $opts)) { if (! $opts['stream']) { throw new Exception( 'Please provide a stream function. Check https://github.com/orhanerday/open-ai#stream-example for an example.' ); } $this->stream_method = $stream; } $opts['model'] = $opts['model'] ?? $this->model; $url = Url::chatUrl(); $this->baseUrl($url); return $this->sendRequest($url, 'POST', $opts); } "
> > Can you help me?
> 
> @df2012 would you please pull the code, I pushed another small fix but seems like it is not related to your issue. please check if you have `baseUrl()` function at the end of `src/OpenAi.php` file

Solved! Thanks bro.

File: README.md [hunk 2]
```diff
@@ -419,6 +419,26 @@ $classification = $open_ai->classification([
...
+## Chat
+
+Given a chat conversation, the model will return a chat completion response.
+
+ ```php
+$complete = $open_ai->chat([
+    'model' => 'gpt-3.5-turbo',
+    'messages' => [
+        [
+            "role" => "user",
```
Comment (oliverschloebe) [inline README.md:431]: Maybe educate the user that `role` can be either “system”, “user”, or “assistant”?

Comment (orhanerday) [reply README.md:431]: I'll make micro-changes after merging for readme, thx.

**Review by orhanerday — APPROVED** [review]
Your work has been an enormous help, and I'm grateful for your time and effort. Thanks, @bashar94!

**Pull request merged**